contracts/erc721/UserMintableERC721.sol

```diff
@@ -525,82 +533,107 @@ contract UserMintableERC721 is Initializable, ERC721, ERC721Enumerable, ERC721Me
     );
   }
 
+  function _requireValidSignatureTransitionProof(
+    bytes32 documentRoot,
+    address identity,
+    bytes32 pbKey,
+    bytes memory transitionValidated,
+    bytes32 transitionValidatedSalt,
+    bytes32[] memory transitionValidatedProof
+  )
+  internal
+  view
+  {
+    // Reconstruct the compact property
+    bytes memory transitionProperty_ = abi.encodePacked(
+      SIGNATURE_TREE_SIGNATURES,
+      identity,
+      pbKey,
+      SIGNATURE_TREE_SIGNATURES_SIGNATURE_TRANSITION
+    );
+
+    // Check transition validated proof
+    require(
+      MerkleProof.verifySha256(
+        transitionValidatedProof,
+        documentRoot,
+        sha256(abi.encodePacked(transitionProperty_, transitionValidated, transitionValidatedSalt))
+      ),
+      "Signature Transition is not valid"
+    );
+  }
+
   /**
    * @dev Checks that provided document is signed by the given identity
    * and validates and checks if the public key used is a valid SIGNING_KEY
-   * @param documentRoot bytes32 the anchored document root
+   * Max number of variables reached
+   * @param b32Values bytes32[] contains [anchored document root, signingRoot, salt, pbKey]
+   * @param btsValues bytes[] contains [signature, transitionValidated]
    * @param anchoredBlock uint32 block number for when the document root was anchored
    * @param identity address Identity that signed the document
-   * @param signingRoot bytes32 hash of all invoice fields which is signed
-   * @param singingRootProof bytes32[] proofs for signing root
-   * @param signature bytes The signature
-   * used to contract the property for precise proofs
-   * @param salt bytes32 salt for leaf construction
-   * @param proof bytes32[] proofs for leaf construction
+   * @param signatureProof bytes32[] proofs for signature
+   * @param signingRootProof bytes32[] proofs for signing root
    */
 
   function _requireSignedByIdentity(
-    bytes32 documentRoot,
+    bytes32[] memory b32Values,
```

Comment:
wrapped common variable types together. It can be improved further by merging as well the two bottom ones, or wrapping them in another struct since is internal.

Reply:
I think is fine as the api of the function is a little bit clearer. Merging salts and proofs will cause a lot of confusion and IMO.

Reply:
Also it is hard to pop stuff from an array and It might make validating the merkle tree more convoluted